src/lib/dataEntities/networks.ts

```diff
@@ -329,9 +349,89 @@ export const l2Networks: L2Networks = {
   },
 }
 
-const getNetwork = async (
+/** Determines if a chain is a parent of *any* other chain. */
+const isParentChain = (chain: Chain): chain is ParentChain => {
+  return (
+    chain &&
+    'partnerChainIDs' in chain &&
+    Array.isArray(chain.partnerChainIDs) &&
+    chain.partnerChainIDs.length > 0
+  )
+}
+
+/** Determines if a chain is a child of *any* other chain. */
+const isChildChain = (chain: Chain): chain is ChildChain => {
+  return chain && 'partnerChainID' in chain
+}
+
+const isL1Chain = (chain: Chain): chain is L1Network => {
+  return chain && isParentChain(chain) && !isChildChain(chain)
+}
+
+const isL2Chain = (chain: Chain): chain is L2Network => {
+  return chain && isChildChain(chain) && !isOrbitChain(chain)
+}
+
+const isOrbitChain = (chain: Chain): chain is OrbitChain => {
+  return chain && 'isOrbit' in chain && chain.isOrbit
+}
+
+/**
+ * Builds an object that is a list of chains filtered by the provided predicate function indexed by their chain id
+ * @param filterFn - A predicate function to determine if a chain should be included.
+ * @return An object with only the filtered chains.
+ */
+const getChainsByType = <T extends typeof Networks>(
+  filterFn: (chain: Chain) => boolean
+): T => {
+  return Object.entries(Networks).reduce<typeof Networks>(
+    (accumulator, [chainId, chainData]) => {
+      if (filterFn(chainData)) {
+        accumulator[chainId] = chainData
+      }
+      return accumulator
+    },
+    {}
+  ) as T
+}
+
+export const getL1Chains = () => getChainsByType<L1Networks>(isL1Chain)
+export const getL2Chains = () => getChainsByType<L2Networks>(isL2Chain)
+export const getParentChains = () =>
+  getChainsByType<ParentChains>(isParentChain)
+export const getChildChains = () => getChainsByType<ChildChains>(isChildChain)
+export const getOrbitChains = () => getChainsByType<OrbitChains>(isOrbitChain)
+
+export const getParentForNetwork = (chain: Chain) => {
+  if (!isChildChain(chain)) {
+    return undefined
+  }
+  const parentChain = Networks[chain.partnerChainID]
+  if (!parentChain || !isParentChain(parentChain)) {
+    throw new ArbSdkError(
+      `ParentChain ${chain.partnerChainID} not recognized for Chain ${chain.chainID}.`
+    )
+  }
+  return parentChain
+}
+
+export const getChildrenForNetwork = (chain: Chain) => {
+  if (!isParentChain(chain)) {
+    return undefined
+  }
+  const childChains = getChildChains()
+  const children = Object.values(childChains).filter(
+    childChain => childChain.partnerChainID === chain.chainID
+  )
+  return children
+}
+
+export let l1Networks: L1Networks = getL1Chains()
+export let l2Networks: L2Networks = getL2Chains()
+
+export const getNetwork = async (
```

Comment:
`getL1Network` should return a value properly if the passed in provider is a provider for an L2 that is a parent chain (e.g. Arbitrum Goerli)